src/components/common/ColonyActions/ActionsListItem/config/config.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Colony,
+  ColonyActions,
+  ColonyMotions,
+  FormattedAction,
+  UniversalMessageValues,
+} from '~types';
+
+import { mapItemToExpectedFormat } from './helpers';
+
+enum ActionTitleMessageKeys {
+  Amount = 'amount',
+  Direction = 'direction',
+  FromDomain = 'fromDomain',
+  Initiator = 'initiator',
+  NewVersion = 'newVersion',
+  Recipient = 'recipient',
+  ReputationChange = 'reputationChange',
+  ReputationChangeNumeral = 'reputationChangeNumeral',
+  RolesChanged = 'rolesChanged',
+  ToDomain = 'toDomain',
+  TokenSymbol = 'tokenSymbol',
+}
+
+/* Maps actionTypes to message values as found in en-actions.ts */
+const getMessageDescriptorKeys = (
+  actionType: ColonyActions | ColonyMotions,
+) => {
+  switch (true) {
+    case actionType.includes(ColonyActions.Payment):
+      return [
+        ActionTitleMessageKeys.Recipient,
+        ActionTitleMessageKeys.Amount,
+        ActionTitleMessageKeys.TokenSymbol,
+      ];
+    case actionType.includes(ColonyActions.MoveFunds):
+      return [
+        ActionTitleMessageKeys.Amount,
+        ActionTitleMessageKeys.TokenSymbol,
+        ActionTitleMessageKeys.FromDomain,
+        ActionTitleMessageKeys.ToDomain,
+      ];
+    case actionType.includes(ColonyActions.UnlockToken):
+      return [ActionTitleMessageKeys.TokenSymbol];
+    case actionType.includes(ColonyActions.MintTokens):
+      return [
+        ActionTitleMessageKeys.Amount,
+        ActionTitleMessageKeys.TokenSymbol,
+      ];
+    case actionType.includes(ColonyActions.CreateDomain):
+      return [ActionTitleMessageKeys.FromDomain];
+    case actionType.includes(ColonyActions.VersionUpgrade):
+      return [ActionTitleMessageKeys.NewVersion];
+    case actionType.includes(ColonyActions.EditDomain):
+      return [ActionTitleMessageKeys.FromDomain];
+    case actionType.includes(ColonyActions.Recovery):
+      return [ActionTitleMessageKeys.Initiator];
+    case actionType.includes(ColonyActions.EmitDomainReputationPenalty):
+      return [
+        ActionTitleMessageKeys.Recipient,
+        ActionTitleMessageKeys.ReputationChangeNumeral,
+        ActionTitleMessageKeys.ReputationChange,
+      ];
+    case actionType.includes(ColonyActions.EmitDomainReputationReward):
+      return [
+        ActionTitleMessageKeys.Recipient,
+        ActionTitleMessageKeys.ReputationChangeNumeral,
+        ActionTitleMessageKeys.ReputationChange,
+      ];
+    case actionType.includes(ColonyActions.SetUserRoles):
+      return [
+        ActionTitleMessageKeys.RolesChanged,
+        ActionTitleMessageKeys.Direction,
+        ActionTitleMessageKeys.FromDomain,
+        ActionTitleMessageKeys.Recipient,
+      ];
+
+    default:
+      return [];
+  }
+};
+
+/* Returns the correct message values according to the action type. */
+const getActionListItemTitleValues = (
```

Comment:
Before we were passing the [`FormattedMessage`](url) component a massive object with all possible title values. Now we fetch the relevant values, according to the action type, and only pass in those.

All this needs to be maintained is that the mapping above gets updated every time a new message is added at `en-actions`.